contracts/ManagerCore.sol

```diff
@@ -73,57 +79,67 @@ contract ManagerCore is Ownable {
      * Initializes any predeployed factories. Note: This function can only be called by
      * the owner once to batch initialize the initial system contracts.
      *
+     * @param _extensions            List of extensions to add
      * @param _factories             List of factories to add
      */
     function initialize(
+        address[] memory _extensions,
         address[] memory _factories
     )
         external
         onlyOwner
     {
         require(!isInitialized, "ManagerCore is already initialized");
 
+        extensions = _extensions;
         factories = _factories;
 
-        // Loop through and initialize isFactory mapping
+        // Loop through and initialize isExtension and isFactory mapping
+        for (uint256 i = 0; i < _extensions.length; i++) {
+            address extension = _extensions[i];
+            require(extension != address(0), "Zero address submitted.");
+            isExtension[extension] = true;
+            emit ExtensionAdded(extension);
```

Comment:
Emitting `ExtensionAdded` and `FactoryAdded` in the `initialize` call is useful for the Graph and other event consumers. We also emit these events in `addExtensions` and `addFactory`. The key is we emit events on **both** initialization and updates. 

The analogous contract in set-protocol-v2, the `Controller`, [does not emit any events in `initialize`](https://github.com/SetProtocol/set-protocol-v2/blob/master/contracts/protocol/Controller.sol#L138). However, it does emit events on updates, such as `addFactory` and `addModule`. This gives event consumers an incomplete picture of state. 

Similarly, the `SetToken` does not emit [initial default units in the constructor](https://github.com/SetProtocol/set-protocol-v2/blob/master/contracts/protocol/SetToken.sol#L182), but does emit [events on updates](https://github.com/SetProtocol/set-protocol-v2/blob/master/contracts/protocol/SetToken.sol#L243) in `editDefaultPositionUnit()`. Note the [SetTokenCreator also does not emit initial units](https://github.com/SetProtocol/set-protocol-v2/blob/master/contracts/protocol/SetTokenCreator.sol#L38).